meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -248,10 +248,11 @@ IMAGE_CMD_resin-sdcard () {
     # Finally add the fingerprint to vfat partition
     mcopy -i ${WORKDIR}/boot.img -v ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT} ::
 
-    # Create a vfat filesystem for config partition
     CONFIG_BLOCKS=$(LC_ALL=C parted -s ${RESIN_SDIMG} unit b print | awk '/ 5 / { print substr($4, 1, length($4 -1)) / 512 /2 }')
     rm -rf ${WORKDIR}/config.img
-    mkfs.vfat -n "${RESIN_CONFIG_FS_LABEL}" -S 512 -C ${WORKDIR}/config.img $CONFIG_BLOCKS
+    # Create an ext4 filesystem for config partition
+    dd if=/dev/zero of=${WORKDIR}/config.img seek="${CONFIG_SIZE}" count=$CONFIG_BLOCKS bs=1024
```

Comment:
Very good catch @floion . Indeed there is an issue here. But it is strange cause yocto guys do exactly the same thing which looks wrong 61:image_types.bbclass.